pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -669,6 +669,12 @@ func (c *Controller) syncControlPlane(clusterDeployment *clustop.ClusterDeployme
 		updatedMachineSet.Spec.Replicas = &machineSetSize
 	}
 
+	// Ensure labels and taints are up to date:
+	for k, v := range machineSetConfig.NodeLabels {
```

Comment:
Shouldn't we just set the Labels to whatever is in the machineSetConfig? I don't think we want to support additional labels in the derived resource.

Reply:
Can we make that assumption about a kube object? I expected they would be a mechanism for arbitrary apps to slap data into, and it wouldn't be safe for us to assume ownership.

Reply:
But this is not the kube object yet... this is just the machineset that was created from the clusterdeployment. In this case, it should be exactly the same.

Reply:
My mistake I follow you now. Thanks.

Reply:
Ok after staring at this for longer than I'd like to admit (there's an awful lot of labels flying around), I think this is out of sync with the code below, and if we change it to just take the NodeLabels wholesale, we introduce a bug. Below, during creation of the machine set, we transfer all NodeLabels, as well as add three for the cluster deployment, cluster, and machine set names. We should be doing the same here during updates to ensure these stay set correctly I'm thinking.